File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/activity/PolarCaloriesData.kt
package com.polar.sdk.api.model.activity

import java.time.LocalDate

/**
 * Activity calories data for a given [date].
 * Includes the calories value based on the requested type (e.g., activity, training, or BMR).
 */
data class PolarCaloriesData(val date: LocalDate? = null, val calories: Int? = null)

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/activity/PolarCaloriesData.swift
///  Copyright © 2025 Polar. All rights reserved.

import Foundation

public struct PolarCaloriesData: Codable {
    public let date: Date
    public let calories: Int
}

public enum CaloriesType: CaseIterable {
    case activity
    case training
    case bmr

    public var displayName: String {
        switch self {
            case .activity: return "Activity"
            case .training: return "Training"
            case .bmr: return "BMR"
        }
    }

    public static var allCases: [CaloriesType] {
        return [.activity, .training, .bmr]
    }
}